lib/common/deprecate.ts

```diff
@@ -84,8 +68,7 @@ export function event (emitter: NodeJS.EventEmitter, oldName: string, newName: s
 // remove a property with no replacement
 export function removeProperty<T extends Object, K extends (keyof T & string)>(object: T, removedName: K, onlyForValues?: any[]): T {
   // if the property's already been removed, warn about it
-  // eslint-disable-next-line no-proto
-  const info = Object.getOwnPropertyDescriptor((object as any).__proto__, removedName);
+  const info = Object.getOwnPropertyDescriptor(object, removedName);
```

Comment:
The use of `object.__proto__` here was causing the tests for `removeProperty` to be silent failures - unfortunately the tests were only checking for the presence of the property name in the logged message but the failure cases here all start with `Unable to remove property '${removedName}'` so that let the tests pass without testing what the intended to test.

This was originally added in #22337, but it's unclear to me what required its usage. We don't have any current usage of `deprecate.removeProperty` in the codebase so if removing the usage of `object.__proto__` here proves to be a problem for future usage it can be restored.